lib/metrics.py

```diff
@@ -12,11 +12,29 @@
 from common import get_media
 from framereader import FrameReaders
 
-def md5(filename):
+def md5(filename, chunksize = 4096, numbytes = -1):
+  if numbytes < 0: # calculate checksum on entire file
+    numbytes = os.stat(filename).st_size
+
+  numbytesread = 0
   m = hashlib.md5()
   with open(filename, "rb") as f:
-    for chunk in iter(lambda: f.read(4096), b""):
+
+    # read numchunks of size chunksize
+    numchunks = int(numbytes / chunksize)
+    for n, chunk in enumerate(iter(lambda: f.read(chunksize), b""), 1):
```

Comment:
if numchunks = 0, it will calculate wrong, how about this.
for n, chunk in enumerate(iter(lambda: f.read(chunksize), b"")):
  if n == numchunks:
    break
  numbytesread += len(chunk)
  m.update(chunk)

Reply:
You are right, we need to handle `numchunks = 0`... good catch.

However, with your suggested change, we would lose the first `chunksize` bytes.  That is, if `numbytes = 4095` and `chunksize = 4096`, then we still read `4096 bytes` from the file in the first loop iteration.  Thus, the `4095 bytes` read for `lastchunk` will be after the first `4096 bytes`.

What if we only do the loop if `numchunks > 0`? For example:
```
if numchunks > 0:
  for n, chunk in ...
lastchunk = ...
```

Reply:
if numbytes = 4095 and chunksize = 4096, then numchunks =0, it will break out the cycle because the n starts from 0 now, it is same with you.

Reply:
add the condition "if numchunks > 0:", it also looks good to me.

Reply:
if `numchunks = 0`, we still enter the loop once while reading chunksize from the file.  I will update to check the `numchunks > 0` condition

Reply:
patch updated.

Reply:
> if `numchunks = 0`, we still enter the loop once while reading chunksize from the file. I will update to check the `numchunks > 0` condition

Maybe the website display problem -_- -_- , I have put "break" condition before "reading chunksize from the file", it won't loop once when numchunks = 0.  -_- -_-

Reply:
> patch updated.

look good to me.

Reply:
> > if `numchunks = 0`, we still enter the loop once while reading chunksize from the file. I will update to check the `numchunks > 0` condition
> 
> Maybe the website display problem -_- -_- , I have put "break" condition before "reading chunksize from the file", it won't loop once when numchunks = 0. -_- -_-

sorry, you are right, it will does loop once, it is my mistake.

Reply:
> > if `numchunks = 0`, we still enter the loop once while reading chunksize from the file. I will update to check the `numchunks > 0` condition
> 
> Maybe the website display problem -_- -_- , I have put "break" condition before "reading chunksize from the file", it won't loop once when numchunks = 0. -_- -_-

Ok, it could be a possible web rendering issue that made it hard to read.  You should style sample code between backticks, \`,  to get better rendering (https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code).

Thanks for reviewing!

Reply:
I dropped into the "for" cycle,  I suddenly thought that we said two different things, you said that it will loop once for reading file, I said that it won't loop once for "m.update(chunk)". So you are right, and it is right for adding the condition "if numchunks > 0:". It is my mistake, sorry for the noise.